src/commands/scaling/addScaleRule/ScaleRuleNameStep.ts

```diff
@@ -3,40 +3,58 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import type { ScaleRule } from '@azure/arm-appcontainers';
+import { ContainerApp, ContainerAppsAPIClient, KnownActiveRevisionsMode, Revision, ScaleRule } from '@azure/arm-appcontainers';
 import { AzureWizardPromptStep } from '@microsoft/vscode-azext-utils';
+import { createContainerAppsAPIClient } from '../../../utils/azureClients';
 import { localize } from '../../../utils/localize';
 import type { IAddScaleRuleContext } from './IAddScaleRuleContext';
 
 export class ScaleRuleNameStep extends AzureWizardPromptStep<IAddScaleRuleContext> {
     public hideStepCount: boolean = true;
 
     public async prompt(context: IAddScaleRuleContext): Promise<void> {
-        context.ruleName = (await context.ui.showInputBox({
+        context.newRuleName = (await context.ui.showInputBox({
             prompt: localize('scaleRuleNamePrompt', 'Enter a name for the new scale rule.'),
-            validateInput: (name: string | undefined): string | undefined => {
-                return validateScaleRuleInput(name, context.containerApp?.name, context.scaleRules);
-            }
+            validateInput: this.validateInput,
+            asyncValidationTask: (name: string) => this.validateNameAvailable(context, name)
         })).trim();
     }
 
     public shouldPrompt(context: IAddScaleRuleContext): boolean {
-        return context.ruleName === undefined;
+        return !context.newRuleName;
     }
-}
 
-function validateScaleRuleInput(name: string | undefined, containerAppName: string, scaleRules: ScaleRule[]): string | undefined {
-    name = name ? name.trim() : '';
+    private validateInput(name: string | undefined): string | undefined {
+        name = name ? name.trim() : '';
+
+        if (!/^[a-z0-9]([-a-z0-9]*[a-z0-9])?$/.test(name)) {
+            return localize('invalidChar', `A name must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character.`);
+        }
 
-    if (!/^[a-z0-9]([-a-z0-9]*[a-z0-9])?$/.test(name)) {
-        return localize('invalidChar', `A name must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character.`);
+        return undefined;
     }
 
-    const scaleRuleExists: boolean = !!scaleRules?.some((rule) => {
-        return rule?.name?.length && rule?.name === name;
-    });
-    if (scaleRuleExists) {
-        return localize('scaleRuleExists', 'The scale rule "{0}" already exists in container app "{1}". Please enter a unique name.', name, containerAppName);
+    private async validateNameAvailable(context: IAddScaleRuleContext, name: string): Promise<string | undefined> {
```

Comment:
Added validation that always does a network call to ensure we aren't working with stale data when we run our verification logic.